[QA] 날짜 피커, 날짜 형식 수정

stopstone opened the pull request:
## 작업 내용
캘린더의 색상을 키 컬러에 맞게 수정합니다.
날짜 형식 변경
앱 내 폰트 스케일 고정

## 확인 방법
앱 내 폰트가 변경되어 간격이 일정하지 않는 부분을 수정했습니다. Theme.kt
캘린더의 텍스트버튼, DatePicker 색상을 키 컬러에 맞게 수정하였습니다.NearDatePicker.kt
yyyy-mm-dd 형태의 날짜 형식을 yyyy.mm.dd 로 변경하였습니다. ContactItem.kt

## 참고 사항
프로필 수정 화면에서 텍스트와 텍스트필드 사이 간격이 맞지 않는 현상이 있어
Theme.kt에서 폰트 스케일을 1f로 고정하였습니다.

stopstone committed:
chore: NearDatePicker 테마 색상 적용

- NearDatePicker의 선택된 날짜, 오늘 날짜, 확인/닫기 버튼 등에 NearTheme의 `BLUE01_5AA2E9` 색상을 적용하여 디자인을 통일했습니다.

stopstone committed:
style: 마지막 연락일 날짜 형식 변경

- 마지막으로 연락한 날짜 형식을 'yyyy-MM-dd'에서 'yyyy.MM.dd'로 변경하여 표시합니다.

stopstone committed:
fix: 앱 내 폰트 스케일 고정

- 사용자가 시스템 폰트 크기를 변경하더라도 앱의 UI가 깨지지 않도록 `fontScale`을 1f로 고정합니다.

stopstone committed:
feat: 연락 주기 표시 방식 구체화

- 연락 주기를 표시할 때, '매주/2주마다'는 요일을, '매달/6개월마다'는 날짜를 함께 보여주도록 수정했습니다.
- `DateExtension`에 오늘 날짜(`getTodayDayOfMonth`)를 가져오는 함수를 추가했습니다.
- `buildContactFrequencyText` 컴포저블 함수를 추가하여 연락 주기 텍스트 생성 로직을 분리했습니다.

stopstone committed:
refactor: 날짜 형식 변환 로직을 Mapper로 이동

- `ContactItem` 컴포저블에 있던 날짜 형식(`yyyy-MM-dd` -> `yyyy.MM.dd`) 변환 로직을 `FriendSummaryMapper`로 이동시켜 데이터 계층에서 처리하도록 변경했습니다.

rhkrwngud445 commented on Near/app/src/main/java/com/alarmy/near/presentation/ui/theme/Theme.kt:
```diff
@@ -29,7 +31,14 @@ fun NearTheme(
             darkTheme -> lightColor // TODO DarkTheme 추가시 수정
             else -> lightColor
         }
+    val currentDensity = LocalDensity.current
+    val themeDensity =
+        Density(
+            density = currentDensity.density,
+            fontScale = 1f,
+        )
```
이 코드는 시스템 폰트 사이즈를 고정시키는 것이 맞을까요?
만약 그렇다면 접근성과 트레이드 오프긴 해서 고민이 되네요🤔

현재로서는 좋아보입니다! TODO로 접근성 관련 수정을 기록해주시면 향후 대응에 좋아보여요!

stopstone replied:
기념일 수정 부분(FriendProfileEditorScreen)에서 
연락 주기 -[패딩]- 텍스트필드 의 유동적인 패딩을 통일하는 방법이 떠오르지 않아 이렇게 구성하였습니다.
접근성 부분에서 아쉬운 부분이 많이 있을 것 같네요 😭

다른 방법으로는 텍스트필드를 고정값으로 해야 될 것 같아 이 부분은 다른 분들과 논의해보면 좋을 것 같습니다!
말씀주신대로 TODO 작성 해두겠습니다!

rhkrwngud445 commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/friendprofileedittor/FriendProfileEditorScreen.kt:
```diff
@@ -613,3 +609,29 @@ fun FriendProfileEditorScreenPreview() {
+        stringResource(
+            R.string.friend_profile_editor_contact_period_format,
+            suffix,
+        )
```
이것도 사소한 거지만 아래와 같은 문법은 어떠실까요?
```kotlin
return "$intervalText ${stringResource(
        R.string.friend_profile_editor_contact_period_format,
        suffix,
    )}" 
``

반영은 안하셔도 됩니다!

stopstone replied:
앗.... 스트링 템플릿을 잘 활용해야겠군요..!
너무 간결하게 되어서 반영안 할 수 없을 것 같습니다 감사합니다!

rhkrwngud445 reviewed (commented):
고생하셨습니다! 
문서화 및 사소한 문법 관련해서 코멘트 남겼습니다! 
확인 부탁드려요😀

stopstone committed:
chore: 시스템 글꼴 크기 고정 관련 TODO 주석 추가

시스템 설정에서 텍스트 크기를 조정해도 앱 내에서는 변경되지 않도록 글꼴 크기를 고정하는 코드에 대한 주석을 추가했습니다. 이로 인해 접근성이 저하될 수 있으므로, 추후 다른 방법으로 텍스트 크기를 조정해야 함을 명시했습니다.

stopstone committed:
refactor: 연락 주기 포맷 스트링 템플릿으로 수정

stopstone committed:
Merge branch 'dev' into qa/calendar-primary-color
